### PR TITLE
Issue 10903 add to role per site

### PR DESCRIPTION
**Pull request by nollymar**

### CODE REVIEW

File: src/com/dotcms/plugin/saml/v3/OpenSamlAuthenticationServiceImpl.java [hunk 1]
```diff
@@ -326,6 +351,23 @@ private String toString(String[] rolePatterns) {
         return null == rolePatterns? NULL : Arrays.asList(rolePatterns).toString();
     }
 
+    private void addRole(final User user, final String roleKey, final boolean createRole, final boolean isSystem) throws DotDataException{
+
+        Role role = this.roleAPI.loadRoleByKey(roleKey);
+
+        //create the role, in case it does not exist
+        if (role == null && createRole){
+            role = createNewRole(roleKey, isSystem);
```
Comment (jdotcms) [inline src/com/dotcms/plugin/saml/v3/OpenSamlAuthenticationServiceImpl.java]: Might be a log here to know that a role has been created could be good.

File: src/com/dotcms/plugin/saml/v3/hooks/SamlHostPostHook.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+import java.io.IOException;
+import java.util.List;
+
+/**
+ * Contains the logic that updates the saml configuration for a host after editing it
+ * Created by nollymar on 3/17/17.
+ */
+public class SamlHostPostHook extends ContentletAPIPostHookAbstractImp {
+
+    private HostAPI hostAPI = APILocator.getHostAPI();
```
Comment (jdotcms) [inline src/com/dotcms/plugin/saml/v3/hooks/SamlHostPostHook.java]: this should be final

File: src/com/dotcms/plugin/saml/v3/hooks/SamlHostPostHook.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+    public SamlHostPostHook() {
+        super();
+    }
+
+    @Override
+    public void checkin(Contentlet currentContentlet, ContentletRelationships relationshipsData, List<Category> cats,
+                        List<Permission> selectedPermissions, User user, boolean respectFrontendRoles,
+                        Contentlet returnValue) {
+        super.checkin(currentContentlet, relationshipsData, cats, selectedPermissions, user, respectFrontendRoles,
```
Comment (dsilvam) [inline src/com/dotcms/plugin/saml/v3/hooks/SamlHostPostHook.java]: Isn't super's checkin empty? https://github.com/dotCMS/core/blob/3.7.0/src/com/dotmarketing/portlets/contentlet/business/ContentletAPIPostHookAbstractImp.java#L233

File: src/com/dotcms/plugin/saml/v3/init/DefaultInitializer.java [hunk 4]
```diff
@@ -35,10 +53,41 @@
...
+        interceptor.delPostHookByClassName(postHook.getClass().getName());
+        try {
+            interceptor.addPostHook(postHook);
+        } catch (InstantiationException e) {
+            e.printStackTrace();
+        } catch (IllegalAccessException e) {
+            e.printStackTrace();
+        } catch (ClassNotFoundException e) {
+            e.printStackTrace();
+        }
```
Comment (dsilvam) [inline src/com/dotcms/plugin/saml/v3/init/DefaultInitializer.java]: we need to change this. e.printStackTrace() for better logging, and have only one catch | muticatch if the catch block will be the same

**Pull request merged**